components/sup/src/manager/service/hook_runner.rs

```diff
@@ -40,7 +57,28 @@ impl<H> HookRunner<H> where H: Hook + Sync
                      pkg,
                      passwd }
     }
+
+    // TODO (DM): Make `HookRunner` and this method allow arbirary configuration of how to run a
+    // hook. For example, a configurable backoff, timeout, or failure tracking. It would also be
+    // nice to merge this with the healthcheck hook retry logic.
+    pub fn loop_future(&self) -> (FutureHandle, impl Future<Item = (), Error = ()>) {
+        let f = future::loop_fn(self.clone(), |hook_runner| {
+            hook_runner.clone()
+                       .into_future()
+                       .map(move |(exit_value, _duration)| {
+                           if hook_runner.hook.retry(exit_value) {
```

Comment:
I like how you added a `retry` function to the `Hook` trait to allow for the (temporal) future where we can start to generalize this kind of hook behavior, rather than tightly-coupling this to the `post-run` hook specifically. We don't lose  anything, but we have a nice foundation from which to build.

Reply:
It would be helpful to add some logging here (probably `debug!`-level) indicating that we're rerunning the offending hook.